front/app/modules/commercial/flag_inappropriate_content/admin/containers/Setting/index.tsx

```diff
@@ -0,0 +1,108 @@
+import useAppConfiguration from 'hooks/useAppConfiguration';
+import messages from './messages';
+import React, { ReactElement } from 'react';
+import { isNilOrError } from 'utils/helperUtils';
+import Link from 'utils/cl-router/Link';
+import styled from 'styled-components';
+import { IconTooltip, Toggle } from 'cl2-component-library';
+import { injectIntl, FormattedMessage } from 'utils/cl-intl';
+import { InjectedIntlProps } from 'react-intl';
+
+const StyledToggle = styled(Toggle)`
+  margin-right: 15px;
+`;
+
+const Setting = styled.div`
+  margin-bottom: 20px;
+`;
+
+const LabelTitleContainer = styled.div`
+  display: flex;
+`;
+
+const LabelTitle = styled.div`
+  font-weight: bold;
+`;
+
+const ToggleLabel = styled.label`
+  display: flex;
+`;
+
+const LabelDescription = styled.div``;
+const LabelContent = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const StyledIconTooltip = styled(IconTooltip)`
+  margin-left: 0.25rem;
+`;
+
+interface Props {
+  onSettingChange: (settingName: string, settingValue: any) => void;
+}
+
+const FlagInnapropriateContentSetting = ({
+  onSettingChange,
+  intl: { formatMessage },
+}: Props & InjectedIntlProps): ReactElement | null => {
+  const appConfiguration = useAppConfiguration();
+  const flagInnaproperiateContentSetting =
+    !isNilOrError(appConfiguration) &&
+    appConfiguration.data.attributes.settings.flag_inappropriate_content
+      ? appConfiguration.data.attributes.settings.flag_inappropriate_content
+      : undefined;
+
+  if (flagInnaproperiateContentSetting) {
+    const flagInnaproperiateContentEnabled =
+      flagInnaproperiateContentSetting.enabled;
+
+    function handleToggle() {
+      onSettingChange('flag_inappropriate_content', {
+        ...flagInnaproperiateContentSetting,
+        flag_inappropriate_content: !flagInnaproperiateContentEnabled,
+      });
+    }
+
+    return (
+      <Setting>
```

Comment:
It's not a big deal, but as a best practice I always try to include `className={className || ''}` in the top most element of a component so that later on it's easily stylable in a parent component (e.g. StyledFlagInnapropriateContentSetting = styled(FlagInnapropriateContentSetting)). I also by default try to use the name `Container` for the top most element. Just some tips though, feel free to use or not use them 🙂.

Reply:
I agree, good remarks